libraries/rush-lib/src/utilities/objectUtilities.ts

```diff
@@ -114,3 +114,56 @@ function cloneDeepInner<TObject>(obj: TObject, seenObjects: Set<unknown>): TObje
     return obj;
   }
 }
+
+/**
+ * Performs a partial deep comparison between `obj` and `source` to
+ * determine if `obj` contains equivalent property values.
+ */
+export function isMatch<TObject>(obj: TObject, source: TObject): boolean {
+  return obj === source || (typeof obj === typeof source && isMatchInner(obj, source));
+}
+
+function isMatchInner<TObject>(obj: TObject, source: TObject): boolean {
+  let sourceKeys: (string | number)[] | undefined;
+  if (typeof source === 'object' && source !== null && !Array.isArray(source)) {
+    sourceKeys = Object.keys(source);
+  } else if (Array.isArray(source)) {
+    sourceKeys = new Array(source.length);
+    for (let i: number = 0; i < source.length; i++) {
+      sourceKeys[i] = i;
+    }
+  }
```

Comment:
Isn't this just populating an array with numbers up to n? I'm confused what the suggestion is here

Reply:
> Isn't this just populating an array with numbers up to n? 

I think his point is that you could more concisely/efficiently create that list by using the same logic as line 133:

```ts
  if ((typeof source === 'object' && source !== null) || Array.isArray(source)) {
    // For an object, make an array of the keys. For an array, make an array of the integer indexes.
    sourceKeys = Object.keys(source);
  }
```

That said, it does annoy me that in order to "check" an array containing 10,000 items, this code will allocate an array containing 10,000 consecutive integers.  Admittedly, this kind of blatant wastefulness is idiomatic in languages like Python and JavaScript. 😄 

....But I'm pretty sure you would eliminate it by going one step further to do:

```ts
   for (const k of Object.keys(source)) {
```

Reply:
Do you mean instead of running `for (const k of sourceKeys) {`, we just replace the entire sourceKeys variable, and directly access the keys on line 145 with `for (const k of Object.keys(source)) {`?

Reply:
I think this makes sense, I've changed it to use `for (const k of Object.keys(source as object)) {` instead of precalculating the sourceKeys.